packages/block-editor/src/store/reducer.js

```diff
@@ -1189,17 +1193,22 @@ export function isSelectionEnabled( state = true, action ) {
 /**
  * Reducer returning the intial block selection.
  *
- * Currently this in only used to restore the selection after block deletion.
- * This reducer should eventually be removed in favour of setting selection
- * directly.
+ * Currently this in only used to restore the selection after block deletion and
+ * pasting new content.This reducer should eventually be removed in favour of setting
+ * selection directly.
  *
  * @param {boolean} state  Current state.
  * @param {Object}  action Dispatched action.
  *
  * @return {?number} Initial position: -1 or undefined.
  */
 export function initialPosition( state, action ) {
-	if ( action.type === 'SELECT_BLOCK' ) {
+	if (
+		action.type === 'REPLACE_BLOCKS' &&
+		typeof action.initialPosition === 'number'
+	) {
+		return action.initialPosition;
+	} else if ( action.type === 'SELECT_BLOCK' ) {
```

Comment:
Why is the type check needed above, but not here?

Reply:
Aside: you could consolidate those two `if` statements into one that uses `||`.

Reply:
If you insist I will, but I think it's more readable like that with a different code branch for each action :-)

Reply:
I respect your preference, if ESLint doesn’t complain it’s fine as is 😃